python/ray/dag/compiled_dag_node.py

```diff
@@ -148,7 +178,12 @@ class CompiledDAG:
     information.
     """
 
-    def __init__(self, buffer_size_bytes: Optional[int]):
+    def __init__(
+        self,
+        buffer_size_bytes: Optional[int],
+        enable_asyncio: bool = False,
```

Comment:
It feels a little weird to me this has to be defined in the constructor. Normally you can just create a DAG and use *_sync & *_async both. 

I wonder if we can lazily start the background reader/writer when we call execute_* first time instead of defininig it at the end of compile?

Reply:
Hmm I think we can probably handle this better in the future? For now, I think it's better to be explicit. The issue with starting both kinds of readers is that the async reader will create background tasks.

Reply:
handling it later also sgtm!